cmake/CheckDependencies.cmake

```diff
@@ -29,6 +29,7 @@ elseif(UNIX)
         libpthread
         libsvace
         libstdc++
+        libresolv
```

Comment:
This shouldn't happen because we [use](https://github.com/tarantool/tarantool-ee/blob/5ff219582c92c5742c1204900c0ccb15973d68b4/cmake/BuildLDAP.cmake#L80-L94) static libresolv on Linux.

I tried building Tarantool EE statically with ASAN enabled with the following options:

```
CC=clang CXX=clang++ cmake -S . -B build/debug-asan-static -DENABLE_ASAN=ON -DENABLE_UB_SANITIZER=ON -DTEST_BUILD=ON -DBUILD_STATIC_WITH_BUNDLED_LIBS=ON
```

The resulting binary didn't have libresol as a dependency:

```
$ ldd build/debug-asan-static/tarantool/src/tarantool
        linux-vdso.so.1 (0x00007ffcf779e000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f99e2b19000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f99e2800000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f99e92cd000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f99e2400000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f99e930a000)
```

Reply:
This looks like a bug. We need to figure out why this happens and whether we can eliminate this dependency.

Reply:
Filed https://github.com/tarantool/tarantool/issues/9740.